sklearn/decomposition/pca.py

```diff
@@ -367,7 +371,10 @@ def _fit(self, X):
 
         # Handle n_components==None
         if self.n_components is None:
-            n_components = X.shape[1]
+            if self.svd_solver is not 'arpack':
```

Comment:
should use `!=` for string comparison (two string objects may have same text but different id)